rust/sedona-functions/src/st_isclosed.rs

```diff
@@ -0,0 +1,175 @@
+use wkb::reader::Wkb;
+
+use crate::executor::WkbExecutor;
+
+pub fn st_isclosed_udf() -> SedonaScalarUDF {
+    SedonaScalarUDF::new(
+        "st_isclosed",
+        vec![Arc::new(STIsClosed {})],
+        Volatility::Immutable,
+        Some(st_is_closed_doc()),
+    )
+}
+
+fn st_is_closed_doc() -> Documentation {
+    Documentation::builder(
+        DOC_SECTION_OTHER,
+        "Return true if the geometry is closed",
+        "ST_IsClosed (A: Geometry)",
+    )
+    .with_argument("geom", "geometry: Input geometry")
+    .with_sql_example("SELECT ST_IsClosed(ST_GeomFromWKT('LINESTRING(0 0, 1 1, 0 1, 0 0)'))")
+    .build()
+}
+
+#[derive(Debug)]
+struct STIsClosed {}
+
+impl SedonaScalarKernel for STIsClosed {
+    fn return_type(&self, args: &[SedonaType]) -> Result<Option<SedonaType>> {
+        let matcher = ArgMatcher::new(
+            vec![ArgMatcher::is_geometry()],
+            SedonaType::Arrow(DataType::Boolean),
+        );
+
+        matcher.match_args(args)
+    }
+
+    fn invoke_batch(
+        &self,
+        arg_types: &[SedonaType],
+        args: &[datafusion_expr::ColumnarValue],
+    ) -> Result<datafusion_expr::ColumnarValue> {
+        let executor = WkbExecutor::new(arg_types, args);
+        let mut builder = BooleanBuilder::with_capacity(executor.num_iterations());
+
+        executor.execute_wkb_void(|maybe_item| {
+            match maybe_item {
+                Some(item) => {
+                    builder.append_value(invoke_scalar(&item)?);
+                }
+                None => builder.append_null(),
+            }
+            Ok(())
+        })?;
+
+        executor.finish(Arc::new(builder.finish()))
+    }
+}
+
+fn invoke_scalar(item: &Wkb) -> Result<bool> {
+    is_geometry_closed(item).map_err(|e| {
+        datafusion_common::error::DataFusionError::Execution(format!(
+            "Failed to check if geometry is closed: {e}"
+        ))
+    })
+}
+
+fn is_geometry_closed<G: GeometryTrait<T = f64>>(item: G) -> Result<bool> {
+    match item.as_type() {
+        geo_traits::GeometryType::LineString(linestring) => {
+            Ok(linestring.to_line_string().is_closed())
+        }
+        geo_traits::GeometryType::MultiLineString(multilinestring) => {
+            Ok(multilinestring.to_multi_line_string().is_closed())
+        }
+        geo_traits::GeometryType::GeometryCollection(geometry_collection) => geometry_collection
+            .to_geometry_collection()
+            .iter()
+            .try_fold(true, |acc, item| {
+                is_geometry_closed(item).map(|is_closed| acc && is_closed)
+            }),
+        geo_traits::GeometryType::Point(_)
+        | geo_traits::GeometryType::Line(_)
+        | geo_traits::GeometryType::MultiPoint(_)
+        | geo_traits::GeometryType::Polygon(_)
+        | geo_traits::GeometryType::MultiPolygon(_)
+        | geo_traits::GeometryType::Rect(_)
+        | geo_traits::GeometryType::Triangle(_) => Ok(true),
```

Comment:
```suggestion
        geo_traits::GeometryType::Point(_)
        | geo_traits::GeometryType::MultiPoint(_)
        | geo_traits::GeometryType::Polygon(_)
        | geo_traits::GeometryType::MultiPolygon(_) => {
            Ok(!is_geometry_empty(&item).map_err(|e| {
                datafusion_common::error::DataFusionError::Execution(format!(
                    "Failed to check if geometry is closed: {e}"
                ))
            })?)
        }
        _ => sedona_internal_err!("Invalid geometry type"),
```

1) SedonaDB doesn't support `Line`, `Rect`, or `Triangle` geometry types, so I think it's better to omit them and error on the default case (this is what we do in other functions). If we somehow got those geom types, I'd rather not hide that bug.
2) I just played queried PostGIS locally, it seems like it returns `false` for these geometries if it's empty, so we should use the `is_geometry_empty` function Sedona has. The import for it is below.

```rust
use sedona_geometry::is_empty::is_geometry_empty;
use datafusion_common::error::DataFusionError
```